daemon/oci_linux.go

```diff
@@ -331,7 +331,7 @@ func WithNamespaces(daemon *Daemon, c *container.Container) coci.SpecOpts {
 		case pidMode.IsContainer():
 			pc, err := daemon.getPidContainer(c)
 			if err != nil {
-				return err
+				return errdefs.InvalidParameter(errors.Wrap(err, "failed to join PID namespace"))
```

Comment:
Shouldn't we return `errdefs.InvalidParameter` inside the `getPidContainer`?

`getPidContainer` can also return `errNotRunning` and `errContainerIsRestarting` which are `errdefs.Conflict`, so we wrap them again in `InvalidParameter` there.

Reply:
Good call. Yes, probably need to make `getPidContainer()` fully responsible for typing the error. Let me update that part.

Note that ultimately, it would all result in an "internal system error", as nothing here can be caused by an invalid request; https://github.com/moby/moby/blob/64f5d9b119bbe95c89c07ed2f5fb4ba87f8095cc/daemon/start.go#L160-L173